python/tests/api/test_partition_spec.py

```diff
@@ -0,0 +1,84 @@
+                        NestedField.required(6, "dec", DecimalType.of(9, 2)),
+                        NestedField.required(7, "s", StringType.get()),
+                        NestedField.required(8, "u", UUIDType.get()),
+                        NestedField.required(9, "f", FixedType.of_length(3)),
+                        NestedField.required(10, "b", BinaryType.get()))
+        specs = [PartitionSpec.builder_for(schema).identity("i").build(),
+                 PartitionSpec.builder_for(schema).identity("l").build(),
+                 PartitionSpec.builder_for(schema).identity("d").build(),
+                 PartitionSpec.builder_for(schema).identity("t").build(),
+                 PartitionSpec.builder_for(schema).identity("ts").build(),
+                 PartitionSpec.builder_for(schema).identity("dec").build(),
+                 PartitionSpec.builder_for(schema).identity("s").build(),
+                 PartitionSpec.builder_for(schema).identity("u").build(),
+                 PartitionSpec.builder_for(schema).identity("f").build(),
+                 PartitionSpec.builder_for(schema).identity("b").build(),
+                 PartitionSpec.builder_for(schema).bucket("i", 128).build(),
+                 PartitionSpec.builder_for(schema).bucket("l", 128).build(),
+                 PartitionSpec.builder_for(schema).bucket("d", 128).build(),
+                 PartitionSpec.builder_for(schema).bucket("t", 128).build(),
+                 PartitionSpec.builder_for(schema).bucket("ts", 128).build(),
+                 PartitionSpec.builder_for(schema).bucket("dec", 128).build(),
+                 PartitionSpec.builder_for(schema).bucket("s", 128).build(),
+                 # todo support them
+                 # PartitionSpec.builder_for(schema).bucket("u", 128).build(),
+                 # PartitionSpec.builder_for(schema).bucket("f", 128).build(),
+                 # PartitionSpec.builder_for(schema).bucket("b", 128).build(),
+                 PartitionSpec.builder_for(schema).year("d").build(),
+                 PartitionSpec.builder_for(schema).month("d").build(),
+                 PartitionSpec.builder_for(schema).day("d").build(),
+                 PartitionSpec.builder_for(schema).year("ts").build(),
+                 PartitionSpec.builder_for(schema).month("ts").build(),
+                 PartitionSpec.builder_for(schema).day("ts").build(),
+                 PartitionSpec.builder_for(schema).hour("ts").build(),
+                 PartitionSpec.builder_for(schema).truncate("i", 10).build(),
+                 PartitionSpec.builder_for(schema).truncate("l", 10).build(),
+                 PartitionSpec.builder_for(schema).truncate("dec", 10).build(),
+                 PartitionSpec.builder_for(schema).truncate("s", 10).build(),
+                 # todo support them
+                 # PartitionSpec.builder_for(schema).add_without_field_id(6, "dec_unsupported", "unsupported").build(),
```

Comment:
Is this not supported?

Reply:
Yep, put todos for unimplemented Transforms, e.g. `BucketUUID`, etc. Also, need to update `from_string` to support unrecognized cases. So I think it might be better to have them in a different PR.